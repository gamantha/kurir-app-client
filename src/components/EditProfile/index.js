import React, { Component } from 'react';
import { 
    Animated,
    Text,
    TextInput,
    View,
    Image,
    Button,
    StyleSheet
} from 'react-native';
import { images } from '../../assets';
import Icon from 'react-native-vector-icons/FontAwesome';

class EditProfile extends Component {
    render() {
        return (
            <View style={styles.container}>
                <View style={styles.boxContainer}>
                    <View style={[styles.boxIcon]}>
                        <View>
                            <Icon size={20} color="#333" name="arrow-left"/>
                        </View>
                        <View>
                            <Icon size={20} color="#BD303f" name="check"/>
                        </View>
                    </View>
                    <View style= {styles.boxTextChange}>
                        <Text style= {styles.textChange}>
                            Edit Profile
                        </Text>
                    </View>
                    <View style={styles.boxImage}>
                        <View style={styles.images}>
                            <Image
                                source={images.google}
                                style={{width: 80, height:80, borderRadius:10}}
                            />
                        </View>
                        <View
                            style={{
                            }} >
                            <Text style={{ textAlign: 'center', color: '#e65659'}}>
                                Change Foto
                            </Text>
                        </View>
                    </View>
                    <View style= {styles.boxInput}>
                        <View style={{flex: 0.7}}>
                            <View style={{
                                flex: 0.2,
                                flexDirection: 'row',
                                alignItems: 'flex-end'
                            }} >
                                <Text style={{ fontSize: 16 }}>Name</Text>
                            </View>
                            <View style={{
                                    flex: 0.5,
                                    justifyContent: 'space-around'
                                }}>
                                    <TextInput
                                        style={[styles.inputText, { flex: 0.7 }]}
                                        autoCapitalize="none"
                                        underlineColorAndroid="transparent"
                                    />
                            </View>

                            <View style={{
                                flex: 0.2,
                                flexDirection: 'row',
                                alignItems: 'flex-end'
                            }} >
                                <Text style={{ fontSize: 16 }}>Email Address</Text>
                            </View>
                            <View style={{
                                    flex: 0.5,
                                    justifyContent: 'space-around'
                                }}>
                                    <TextInput
                                        style={[styles.inputText, { flex: 0.7 }]}
                                        autoCapitalize="none"
                                        underlineColorAndroid="transparent"
                                    />
                            </View>

                            <View style={{
                                flex: 0.2,
                                flexDirection: 'row',
                                alignItems: 'flex-end'
                            }} >
                                <Text style={{ fontSize: 16 }}>Phone Number</Text>
                            </View>
                            <View style={{
                                    flex: 0.5,
                                    justifyContent: 'space-around'
                                }}>
                                    <TextInput
                                        style={[styles.inputText, { flex: 0.7 }]}
                                        autoCapitalize="none"
                                        underlineColorAndroid="transparent"
                                        placeholder="+62"
                                        placeholderTextColor="#4c4c4c"
                                    />
                            </View>

                            <View style={{
                                flex: 0.2,
                                flexDirection: 'row',
                                alignItems: 'flex-end'
                            }} >
                                <Text style={{ fontSize: 16 }}>Address</Text>
                            </View>
                            <View style={{
                                    flex: 0.5,
                                    justifyContent: 'space-around'
                                }}>
                                    <TextInput
                                        style={[styles.inputText, { flex: 0.7 }]}
                                        autoCapitalize="none"
                                        underlineColorAndroid="transparent"
                                    />
                            </View>

                        </View>
                    </View>
                </View>
            </View>   
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#FFFFFF',
    },
    boxContainer: {
        flex:1,
        marginLeft: 25,
        marginRight: 25,
    },

    boxIcon: {
        flexDirection:'row',
        justifyContent: 'space-between',
        marginTop: 10
    },

    boxTextChange: {
        flex:0.1,
        marginTop:10,
        marginBottom:20,
        flexDirection: 'row',
        alignItems: 'flex-start'
    },

    textChange: {
        fontSize: 16,
        fontWeight: 'bold',
        color: '#333'
    },

    boxImage: {
        flexDirection: 'column',
    },

    images: {
        justifyContent: 'center',
        alignItems: 'center'
    },

    boxInput: {
        flex:1,
        marginTop:20
    },

    inputText: {
        flex: 1,
        height: 20,
        borderWidth: 1,
        borderColor: '#BD303f',
        borderRadius: 50,
        paddingLeft: 20
    }
    
});
export default EditProfile;