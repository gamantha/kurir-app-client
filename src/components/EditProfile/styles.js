import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'space-between',
        backgroundColor: '#FFFFFF',
        paddingLeft: 16,
        paddingRight: 16,
    },
    header: {
        fontWeight: "400",
        fontSize: 16,
        letterSpacing: 3,
    },
    inputTextContainer: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    },
    textTitle: {
        paddingTop: 12,
        paddingBottom: 5
    },
    inputText: {
        flex: 1,
        height: 50,
        borderWidth: 1,
        borderColor: '#BD303f',
        borderRadius: 40,
        paddingLeft: 20,
    },
    icon: {
        marginRight: 5
    },
    boxImage: {
        flexDirection: 'column',
    },

    images: {
        justifyContent: 'center',
        alignItems: 'center'
    },
});
