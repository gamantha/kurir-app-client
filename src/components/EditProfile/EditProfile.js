import React, { Component } from 'react';
import { 
    Animated,
    Text,
    TextInput,
    View,
    Image,
    Button,
    StyleSheet,
    TouchableOpacity
} from 'react-native';
import { images } from '../../assets';
import Icon from 'react-native-vector-icons/FontAwesome';
import globalStyles from '../../helpers/styles';
import styles from "./styles";

class EditProfile extends Component {
    render() {
        return (
                <View style={[styles.container]}>
                    <Text style={styles.header}>Edit Profile</Text>
                    <View style={styles.boxImage}>
                        <View style={styles.images}>
                            <Image
                                source={images.profile}
                                style={{width: 80, height:80, borderRadius:10}}
                            />
                        </View>
                        <View
                            style={{
                            }} >
                            <Text style={{ textAlign: 'center', color: '#e65659'}}>
                                Change Foto
                            </Text>
                        </View>
                    </View>
                    <View
                        style={{
                            flex: 0.8,
                            flexDirection: 'column',
                            justifyContent: 'space-between'
                        }}
                    >
                        <Text style={styles.textTitle}>Name</Text>
                        <View style={styles.inputTextContainer}>
                            <TextInput
                                placeholder="Enter your name"
                                style={[
                                    styles.inputText
                                ]}
                                underlineColorAndroid="transparent"
                            />
                        </View>
                    </View>
                    <View
                        style={{
                            flex: 0.8,
                            flexDirection: 'column',
                            justifyContent: 'space-between'
                        }}
                    >
                        <Text style={styles.textTitle}>Phone Number</Text>
                        <View style={styles.inputTextContainer}>
                            <TextInput
                                placeholder="Enter your phone number"
                                style={[
                                    styles.inputText
                                ]}
                                underlineColorAndroid="transparent"
                            />
                        </View>
                    </View>
                    <View
                        style={{
                            flex: 2,
                            flexDirection: 'column',
                            justifyContent: 'space-between'
                        }}
                    >

                        <TouchableOpacity
                            style={[
                                globalStyles.touchAbleButton,
                                { height: 40, marginBottom: 20, marginTop:20 }
                            ]}
                            disabled={false}
                        >
                            <Text style={globalStyles.textButton}>Edit Profile</Text>
                        </TouchableOpacity>
                    </View>
                </View>
        )
    }
}

export default EditProfile;