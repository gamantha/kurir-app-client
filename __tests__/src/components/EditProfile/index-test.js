import React from 'react';
import EditProfile from '../../../../src/components/EditProfile';
import renderer from 'react-test-renderer';

test('EditProfile renders correctly', () => {
    const tree = renderer.create(<EditProfile />).toJSON();
    expect(tree).toMatchSnapshot();
});
