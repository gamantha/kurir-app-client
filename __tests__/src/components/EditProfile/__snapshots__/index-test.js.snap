// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 25,
        "marginRight": 25,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 10,
          },
        ]
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#333",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#BD303f",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 0.1,
          "flexDirection": "row",
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Edit Profile
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/googleplus.png",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 80,
              "width": 80,
            }
          }
        />
      </View>
      <View
        style={Object {}}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e65659",
              "textAlign": "center",
            }
          }
        >
          Change Foto
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.7,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.2,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Name
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.5,
              "justifyContent": "space-around",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            style={
              Array [
                Object {
                  "borderColor": "#BD303f",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 20,
                  "paddingLeft": 20,
                },
                Object {
                  "flex": 0.7,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.2,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Email Address
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.5,
              "justifyContent": "space-around",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            style={
              Array [
                Object {
                  "borderColor": "#BD303f",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 20,
                  "paddingLeft": 20,
                },
                Object {
                  "flex": 0.7,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.2,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Phone Number
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.5,
              "justifyContent": "space-around",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            placeholder="+62"
            placeholderTextColor="#4c4c4c"
            style={
              Array [
                Object {
                  "borderColor": "#BD303f",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 20,
                  "paddingLeft": 20,
                },
                Object {
                  "flex": 0.7,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.2,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Address
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0.5,
              "justifyContent": "space-around",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            style={
              Array [
                Object {
                  "borderColor": "#BD303f",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 20,
                  "paddingLeft": 20,
                },
                Object {
                  "flex": 0.7,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
